x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ForceMergeAction.java

```diff
@@ -131,6 +133,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         StepKey forceMergeKey = new StepKey(phase, NAME, ForceMergeStep.NAME);
         StepKey countKey = new StepKey(phase, NAME, SegmentCountStep.NAME);
 
+        BranchingStep conditionalSkipShrinkStep = new BranchingStep(preForceMergeBranchingKey, checkNotWriteIndex, nextStepKey,
+            (index, clusterState) -> clusterState.getMetadata().index(index).getSettings().get("index.store.snapshot.index_name") != null);
```

Comment:
@dakrone there are several setting we configure for a searchable snapshot. I wonder if we should be more strict here and check the index name? I chose not to as that's an invariant that should be guaranteed by the `searchable-snapshots` module.

What do you think?